src/test/java/rs/hybridit/challenges/ControllerTest.java

```diff
@@ -0,0 +1,151 @@
+package rs.hybridit.challenges;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.mockito.stubbing.Answer;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import rs.hybridit.controller.LibraryController;
+import rs.hybridit.dto.LibraryDto;
+import rs.hybridit.model.Library;
+import rs.hybridit.service.LibraryService;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.mockito.Matchers.isA;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.verifyZeroInteractions;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ControllerTest {
+
+	private LibraryController libraryController;
+
+	@Mock
+	private LibraryService libraryService;
+
+	@Before
+	public void init() {
+		libraryController = new LibraryController(libraryService);
```

Comment:
No need for initialising library controller, since it is the class that is tested, just annotate it with @InjectMocks like this:

@InjectMocks
private LibraryController libraryController;